cb-talent-development/employer-style-base
# CE-495: Hero Carousel (ESB)

**Purpose**
> As a PO, I need a highly flexible hero carousel that displays on desktop and tablet, so I can update slides, copy, images and gradients without the assistance of a developer.

This PR introduces a new kind of header for use with the site. The first page that will display this header is the homepage. There are several new, generic, mixins and associated styles introduced in our style base to control this kind of header. This header is not dependent on any page, it can be added to our interior pages as well.

These new header mixins are specific to carousel headers, with generic styles being added into our Selectors folder to be used with a carousel React component.

**JIRA**
https://careerbuilder.atlassian.net/browse/CE-495

**Changes**
* Improvements and fixes
  * Replaces the homepage header hero with a carousel
  * Cleans up home-page.scss to make way for this new type of header
  * New simplified header styles introduced in style base that could contribute to our Header cleanup epic that Stephanie created

* Changes to developer setup/environment
  * N/A

* Architectural changes
  * N/A

* Migrations or Steps to Take on Production
  * N/A
  
* Library changes
  * N/A

* Side effects
  * N/A

**Screenshots**
* Before
![screen shot 2017-11-08 at 4 36 59 pm](https://user-images.githubusercontent.com/5348098/32578504-1c7b9964-c4a3-11e7-9267-e029ca1f3466.png)

* After
![screen shot 2017-11-08 at 4 37 12 pm](https://user-images.githubusercontent.com/5348098/32578513-204cb690-c4a3-11e7-94d0-aab75bcda040.png)

**Feature Server**
http://web.employer-2.development.c66.me/
http://dev.admin.cbcortex.com/legacy#/webpages (employer-2 homepage)

**How to Verify These Changes**
* Specific pages to visit
  * The home page

* Steps to take
  * View the homepage on your browser and resize your screen. The header should resize appropriately, and you should see a static header for mobile. Compare it to the hiringDot homepage, it should look fairly similar to what is displayed on the hiringDot version.
     * Keep in mind that this header came with no mockup and is supposed to be a highly flexible, generic carousel. So some elements that were specific to the homepage were intentionally left out.
  * Verify the font size against the ACs in the Jira
  * Go to Cortex and begin editing the homepage. Play around with the data structure that you see for the Carousel legacy widget. You should be able to change the following:
     * Gradient (see the style base for the names)
     * Color of the header and subheader (white or blue)
     * Position of the content (left or centered)
     * Remove / add one or two CTA buttons
     * The classes of the CTA buttons
     * Convert a CTA button into a text link, and change the color to white
     * Remove / add a 'learn more' link at the end of the subheader
     * Add a (appropriate) background image
     * The number of slides

* Responsive considerations
  * Despite all flexibility, the header should respond to the browser. Only slide one of the carousel should display in mobile and have appropriate styling similar to what is see on hiring dot.

**Relevant PRs/Dependencies**
  * https://github.com/cbdr/employer/pull/1040

**Additional Information**
N/A
